### ISSUE TITLE
Issue 08: coarse_usable_score head の supervised loss と inference score mode を追加する

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#369]
# Issue 08: coarse_usable_score head の supervised loss と inference score mode を追加する

## 背景

Step 3 では、OOF coarse error から作った label を使い、`coarse_usable_score` を学習したい。これは PU ではなく、manual pick が存在する trace の中で

```text
coarse pick が manual に近いか
```

を教師ありで学習するタスクである。

この head は physics 前段の目的に直接対応する。

## ゴール

coarse training に以下の supervised loss を追加する。

```text
L_total = L_pick_location + λ_pu L_pickability + λ_use L_coarse_usability
```

`coarse_usability.enabled: false` では完全に legacy / Step 2 と同じ。

## 変更対象の候補

- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/loss.py`
- 新規または既存: `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/usability_loss.py`
- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/output.py`
- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/infer.py`
- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/config.py`

### label

Issue 07 により batch meta に以下がある前提。

```python
labels = batch["meta"]["coarse_usable_label_view"]  # B,H
```

値:

- `1`: usable positive
- `0`: usable negative
- `-1`: ignore

### BCE loss

```python
mask = labels != ignore_value
loss = BCEWithLogitsLoss(reduction="none")(logits[mask], labels[mask].float()).mean()
```

要件:

- `trace_valid == False` は必ず ignore。
- label が全て ignore の batch では usability loss を 0 にする。
- usability enabled なのに `coarse_usable_logits` が pred に無い場合はエラー。
- usability enabled なのに label meta が無い場合はエラー。

### class imbalance

usable positive/negative は偏る可能性があるため、config に optional `pos_weight` を入れてもよい。

```yaml
train:
  coarse_usability:
    enabled: true
    weight: 0.2
    pos_weight: null
```

初期実装では `pos_weight` optional。未指定なら通常 BCE。

## inference score mode

Issue 04 の `physics_input_score.mode` に Step 3 mode を実装する。

推奨本命:

```text
physics_input_score = coarse_pickability_score * coarse_usable_score
```

比較用に以下も許可する。

```text
physics_input_score = coarse_pickability_score * coarse_usable_score * coarse_pmax
```

注意:

- `coarse_usable_score` は sigmoid(logits)。
- `coarse_pickability_score` が無い場合、`pickability_times_usability` はエラー。
- `coarse_usable_score` が無い場合もエラー。

## infer epoch metrics

training skeleton は `InferEpochResult(metrics=...)` をサポートしている。可能なら `_run_infer_epoch()` で usability label がある場合だけ以下を返す。

```text
coarse_usability_bce
coarse_usability_acc@0.5
coarse_usability_pos_rate_pred@0.5
coarse_usability_label_coverage
```

この metric は必須ではないが、Step 3 の debug に有用。実装する場合、`ckpt.metric` に指定できるよう key を stable にする。

## 受け入れ条件

- usability disabled では既存 loss と Step 2 loss の挙動が変わらない。
- usability enabled で labels/head が揃っていれば finite scalar loss が返る。
- labels `-1` は loss から除外される。
- labels が全て `-1` の batch でも落ちない。
- inference で `coarse_usable_score` が NPZ に保存される。
- `mode: pickability_times_usability` で `physics_input_score` が期待通りになる。

## テスト

新規テスト候補:

- `tests/test_fbpick_coarse_usability_loss.py`

テスト内容:

1. labels `1/0/-1` の BCE が mask 付きで計算される。
2. all ignore で 0 loss。
3. missing head / missing labels でエラー。
4. `physics_input_score` mode の計算。

## 非ゴール

- OOF label 生成は Issue 06。
- dataset label projection は Issue 07。
- PU pickability loss は Issue 03。
